src/hotspot/share/logging/logFileStreamOutput.cpp

```diff
@@ -115,18 +116,34 @@ bool LogFileStreamOutput::flush() {
   total += result;                                            \
 }
 
-int LogFileStreamOutput::write_internal(const LogDecorations& decorations, const char* msg) {
+int LogFileStreamOutput::write_internal_line(const LogDecorations& decorations, const char* msg, int msg_len) {
   int written = 0;
   const bool use_decorations = !_decorators.is_empty();
 
-  if (use_decorations) {
-    WRITE_LOG_WITH_RESULT_CHECK(write_decorations(decorations), written);
-    WRITE_LOG_WITH_RESULT_CHECK(jio_fprintf(_stream, " "), written);
-  }
-
   if (!_fold_multilines) {
-    WRITE_LOG_WITH_RESULT_CHECK(jio_fprintf(_stream, "%s\n", msg), written);
+    const char* base = msg;
+    int written_msg = 0;
+    int decorator_padding = 0;
+    if (use_decorations) {
+      WRITE_LOG_WITH_RESULT_CHECK(write_decorations(decorations), decorator_padding);
+      WRITE_LOG_WITH_RESULT_CHECK(jio_fprintf(_stream, " "), written);
+    }
+    written += decorator_padding;
+    WRITE_LOG_WITH_RESULT_CHECK(jio_fprintf(_stream, "%s\n", msg), written_msg);
+    while (written_msg < msg_len) {
```

Comment:
Some comments here about how this handles multi-lines, would be good. Something like:
```
// If there is still more to write out then we must have a multi-line message.
// If we have active decorators then pad the line with an empty decorator string so 
// that the output lines up.
```